Python: Bug: Make Install does not work on clean builds

**Describe the bug**
Tried to follow the instructions for [DEV SETUP](https://github.com/microsoft/semantic-kernel/blob/main/python/DEV_SETUP.md#for-mac-and-linux-both-native-and-wsl) on testing semantic-kernel/python. My device did not have `uv` installed and, as a result, I was unable to leverage the makefile flow.

We want to use these steps for our own automated testing of the MongoDB library, so being able to follow the instructions is essential to maintain as simple a testing environment as possible.

**To Reproduce**
Steps to reproduce the behavior:
1. clone the repo
2. Go to `semantic-kernel/python`
3. run `make install`
```
make install
uv found
running uv update
make[1]: uv: No such file or directory
make[1]: *** [install-uv] Error 1
make: *** [install] Error 2
```

**Expected behavior**

I expected the makefile to execute the curl command: `curl -LsSf https://astral.sh/uv/install.sh | sh` and proceed with the next steps of installing. This does not work and I keep getting `uv not found`. After checking, using `which uv` I confirmed uv did not exist within the same shell instance. 

Moreover if I run the curl command myself,  I still cannot make it through the remainder of the script as `uv` is not a recognized command. Ultimately, I ended up installing uv myself and then running the other commands underneath `make install` sequentially thereafter. I assume this is not desired behavior, though.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Platform**
 - OS: [MacOS Sonoma 14.6.1]
 - IDE: [N/A]
 - Language: [Python]
 - Source: [cloned repo]

**Additional context**
Add any other context about the problem here.

--- a/python/Makefile
+++ b/python/Makefile
@@ -43,7 +43,7 @@ ifndef UV_VERSION
 	curl -LsSf https://astral.sh/uv/install.sh | sh
 	echo "uv installed"
 	# Re-exec the shell so the newly installed uv is available immediately
-    exec $$SHELL
+	exec $$SHELL
 else
 	echo "uv found $(UV_VERSION)"
 	echo "running uv update"

--- a/python/Makefile
+++ b/python/Makefile
@@ -1,13 +1,19 @@
 SHELL = /bin/bash
 
-.PHONY: help install clean build
-.SILENT:
+.PHONY: help install install-uv install-python install-pre-commit install-sk clean build
+.SILENT:  # not strictly required, but included per the original example
 all: install
 
 ifeq ($(PYTHON_VERSION),)
-    PYTHON_VERSION="3.10"
+    PYTHON_VERSION = 3.10
 endif
 
+# Detect uv on PATH
+UV_VERSION := $(shell command -v uv 2> /dev/null)
+
+##############################
+#  HELP
+##############################
 .ONESHELL:
 help:
 	echo -e "\033[1mUSAGE:\033[0m"
@@ -15,10 +21,10 @@ help:
 	echo ""
 	echo -e "\033[1mTARGETS:\033[0m"
 	echo "  help                 - show this help message"
-	echo "  install              - install uv, python, Semantic Kernel and all dependencies"
-	echo "      This is the default and will use Python 3.10."
+	echo "  install              - install uv, python, Semantic Kernel, and all dependencies"
+	echo "                        This is the default and will use Python 3.10."
 	echo "  install-uv           - install uv"
-	echo "  install-python       - install python distributions"
+	echo "  install-python       - install multiple python distributions"
 	echo "  install-sk           - install Semantic Kernel and all dependencies"
 	echo "  install-pre-commit   - install pre-commit hooks"
 	echo "  clean                - remove the virtualenvs"
@@ -26,52 +32,72 @@ help:
 	echo ""
 	echo -e "\033[1mVARIABLES:\033[0m"
 	echo "  PYTHON_VERSION       - Python version to use. Default is 3.10"
-	echo "      By default, 3.10, 3.11 and 3.12 are installed as well."
+	echo "                        By default, 3.10, 3.11, and 3.12 are installed as well."
 
+##############################
+#  INSTALL
+##############################
 install:
 	make install-uv
 	make install-python
 	make install-sk
 	make install-pre-commit
 
-UV_VERSION := $(shell command -v uv 2> /dev/null)
+##############################
+#  INSTALL-UV
+##############################
 install-uv:
-# Check if uv is installed
-ifndef UV_VERSION
-	echo "uv could not be found"
-	echo "Installing uv"
-	curl -LsSf https://astral.sh/uv/install.sh | sh
-	echo "uv installed"
-	# Re-exec the shell so the newly installed uv is available immediately
-	exec $$SHELL
+	# If uv is not found AND we haven't already re-invoked with CONTINUE=1...
+ifneq ($(UV_VERSION),)
+	echo "uv found at: $(UV_VERSION)"
+	echo "running uv self update"
+	uv self update
+else ifeq ($(CONTINUE),1)
+	echo "Skipping uv re-install; continuing with the rest of the steps."
 else
-	echo "uv found $(UV_VERSION)"
-	echo "running uv update"
-	uv self update 
+	echo "uv could not be found."
+	echo "Installing uv..."
+	curl -LsSf https://astral.sh/uv/install.sh | sh
+	echo "uv installed."
+	echo "Re-executing shell so uv is immediately available on PATH..."
+	exec $$SHELL -c 'make install CONTINUE=1'
 endif
 
+##############################
+#  INSTALL-PYTHON
+##############################
 .ONESHELL:
 install-python:
 	echo "Installing python 3.10, 3.11, 3.12"
 	uv python install 3.10 3.11 3.12
 
+##############################
+#  INSTALL-PRE-COMMIT
+##############################
 .ONESHELL:
 install-pre-commit:
 	echo "Installing pre-commit hooks"
 	uv run pre-commit install -c python/.pre-commit-config.yaml
 
-
+##############################
+#  INSTALL-SK
+##############################
 .ONESHELL:
 install-sk:
 	echo "Creating and activating venv for python $(PYTHON_VERSION)"
 	uv venv --python $(PYTHON_VERSION)
 	echo "Installing Semantic Kernel and all dependencies"
 	uv sync --all-extras --dev
-	
+
+##############################
+#  CLEAN
+##############################
 .ONESHELL:
 clean:
-	# Remove the virtualenv
 	rm -rf .venv
 
+##############################
+#  BUILD
+##############################
 build:
 	uvx --from build pyproject-build --installer uv

--- a/python/Makefile
+++ b/python/Makefile
@@ -42,8 +42,8 @@ ifndef UV_VERSION
 	echo "Installing uv"
 	curl -LsSf https://astral.sh/uv/install.sh | sh
 	echo "uv installed"
-	echo "Please restart your shell."
-	exit 1
+	# Re-exec the shell so the newly installed uv is available immediately
+    exec $$SHELL
 else
 	echo "uv found $(UV_VERSION)"
 	echo "running uv update"

